### PR TITLE
Dev docs homepage languages

### PR DESCRIPTION
**Pull request by dprats**
made updates to the developer docs, relating to programming languages

### CODE REVIEW

File: docs/developer-docs/backend/choosing-language.md [hunk 1]
```diff
@@ -3,35 +3,41 @@ sidebar_position: 1
...
-Rust is a powerful and modern type-sound programming language with an active developer community.
-Of course, developers are not required to work in one (or both) of those languages. There are many different community-supported languages available as well, as you can see below in the list of available languages.
+To create smart contract canisters, it is typical to use a so-called CDK, a Canister Development Kit.
+The CDK's main tasks are to expose the public interface of a canister, manage memory, allow other canisters to be called, and to interface with the system API. Because the Internet Computer supports dapps compiled to standard WebAssembly modules, you can use many different programming languages to create your canisters.
 
-It is also possible to split your work between multiple languages. Different canister smart contracts talk to each other using the [Candid](./candid/index.md) language. What language works behind the candid interface, however, does not matter.
+In theory, any language that can be compiled into WebAssembly can be used to write smart contract canisters.
```
Comment (dfx-json) [inline docs/developer-docs/backend/choosing-language.md:9]: is it worth mentioning that the produced WASM has to be compiled for an IC specific runtime?

File: docs/developer-docs/backend/choosing-language.md [hunk 2]
```diff
@@ -3,35 +3,41 @@ sidebar_position: 1
...
+## A comparison between Motoko and Rust
+
+To help developers interested in Motoko, here is a comparison of Motoko and Rust (a popular language in Web3). 
+
+As a rule of thumb in deciding between Motoko and Rust:
+
+* Motoko is much easier to learn and ergonomic to use for application developers. It has syntax and rules familiar to developers with a background in the application layer (JavaScript, Ruby, Python, Solidity, etc...). Motoko is good for getting sample, demo, or smaller contracts shipped quickly, but its library ecosystem is still early so can prove challenging for larger projects.
 
-### A comparison between Motoko and Rust
+* Rust is a good place for those who already know Rust, come from a systems engingeering background (C, C++), or are starting larger or more complex prokjects where having a baked library ecosystem is helpful or important.
```
Comment (dfx-json) [inline docs/developer-docs/backend/choosing-language.md:36]: typo: prokjects

Comment (dprats) [reply docs/developer-docs/backend/choosing-language.md]: fixed

File: docs/developer-docs/backend/choosing-language.md (hunk 1 above)
Comment (dprats) [reply docs/developer-docs/backend/choosing-language.md]: i added something. let me know what you think

File: docs/developer-docs/backend/choosing-language.md [hunk 3]
```diff
@@ -3,35 +3,41 @@ sidebar_position: 1
...
-Rust is a powerful and modern type-sound programming language with an active developer community.
-Of course, developers are not required to work in one (or both) of those languages. There are many different community-supported languages available as well, as you can see below in the list of available languages.
+To create smart contract canisters, it is typical to use a so-called CDK, a Canister Development Kit.
+The CDK's main tasks are to expose the public interface of a canister, manage memory, allow other canisters to be called, and to interface with the system API. Because the Internet Computer supports dapps compiled to standard WebAssembly modules, you can use many different programming languages to create your canisters.
 
-It is also possible to split your work between multiple languages. Different canister smart contracts talk to each other using the [Candid](./candid/index.md) language. What language works behind the candid interface, however, does not matter.
+In theory, any language that can be compiled into a WebAssembly module, can produce modules [tailored for the IC](../../references/ic-interface-spec.md) depoyable as ICP smart contract.
```
Comment (dfx-json) [inline docs/developer-docs/backend/choosing-language.md:9]: I like this better; it prevents a potential footgun.

One tiny grammar nit with suggested change:
```suggestion
In theory, any language that can be compiled into a WebAssembly module, can produce modules [tailored for the IC](../../references/ic-interface-spec.md) depoyable as an ICP smart contract.
```

**Review by dfx-json — APPROVED** [review]

File: docs/developer-docs/backend/choosing-language.md [hunk 4]
```diff
@@ -3,35 +3,41 @@ sidebar_position: 1
...
+To create smart contract canisters, it is typical to use a so-called CDK, a Canister Development Kit.
+The CDK's main tasks are to expose the public interface of a canister, manage memory, allow other canisters to be called, and to interface with the system API. Because the Internet Computer supports dapps compiled to standard WebAssembly modules, you can use many different programming languages to create your canisters.
 
-It is also possible to split your work between multiple languages. Different canister smart contracts talk to each other using the [Candid](./candid/index.md) language. What language works behind the candid interface, however, does not matter.
+In theory, any language that can be compiled into a WebAssembly module, can produce modules [tailored for the IC](../../references/ic-interface-spec.md) depoyable as an ICP smart contract.
+
+In practice, the amount of CDK and library support for different anguages varies across the ICP developer ecosystem, so this article lays out common paths for entering developers. The most common languages to use are:
```
Comment (domwoe) [inline docs/developer-docs/backend/choosing-language.md:11]: Small typo: *anguages*

**Pull request merged**